### ISSUE TITLE
Assertion failed in `setInlineInstance`, `XFormsModel.scala`

### ISSUE THREAD

**Issue by avernet** [orbeon/orbeon-forms#5772]
Following #5750. To reproduce:

- In Form Builder, create form with US state field.
- Copy the source.
- Create another form, edit source, paste.

[+1 from customer](https://3.basecamp.com/3600924/buckets/24212679/messages/6057206254)

**Comment by ebruchez** [comment]
In `XXFormsDynamicControl.processFullUpdate()`, we create `InstancesControls(Nil, relevantSwitchState)`. So there are no instances in the "state to restore", apparently on purpose.

Later, `missingInstancesIt` is not empty, so we call `setInlineInstance()` which hits `assert(instance.useInlineContent)`.

However, the missing instance has `useInlineContent == false`. It is readonly-cached:

```xml
<xf:instance src="oxf:/xbl/orbeon/us-state/states.xml" xxf:readonly="true" xxf:cache="true"/>
```

**Comment by ebruchez** [comment]
Original issue is fixed, but reopening because local instances are not correctly restored. For example create form with Dropdown Date, enter value, add new control. This didn't happen before the fix.

**Comment by ebruchez** [comment]
Before the fix for #5750, we were not detecting the case, in `XFormsComponentControl.onCreate()`, where we were "restoring state" properly. This means we were not calling `restoreModels()`, but always `initializeModels()`.

With Form Builder, `xxf:dynamic` also uses this restoration logic. In the case of the "Dropdown Date", this means for example  that the component would act as newly created, so it would take its value from the data, initialize its internal instance, and things would appear ok.

After the fix for #5750, we detect that we are "restoring state", including in Form Builder's `xxf:dynamic`. In `XFormsComponentControl.onCreate()`, we call `restoreModels()`. However, here we use `InstancesControls(instances = Nil, controls = relevantSwitchState))`. That is, Form Builder doesn't restore instance state, only switch state. So the XBL instances are not restored but initialized like new. And at the same time, because we are in the `restoreModels()` and not `initializeModels()`, the XBL models do not initialize as if newly created, so their local instances remains with their default values.

Possible solutions for `xxf:dynamic` only:

1. Save and restore instance state as well.
2. Somehow restore the previous behavior in this case, that is call `initializeModels()`.

Solution 1 seems more correct but questions:

- What is the impact of saving/restoring instance state?
- Are there any possible side-effects?
